lib/txgh/key_manager.rb

```diff
@@ -5,49 +5,91 @@
 module Txgh
   class KeyManager
     class << self
-      def config_from_project(project_name, tx_config = nil)
+      def config_from_project(project_name)
         project_config = project_config_for(project_name)
         repo_config = repo_config_for(project_config['push_translations_to'])
-        tx_config ||= Txgh::TxConfig.load_file(project_config['tx_config'])
-        Txgh::Config.new(project_config, repo_config, tx_config)
+        Txgh::Config.new(project_config, repo_config)
       end
 
-      def config_from_repo(repo_name, tx_config = nil)
+      def config_from_repo(repo_name)
         repo_config = repo_config_for(repo_name)
         project_config = project_config_for(repo_config['push_source_to'])
-        tx_config ||= Txgh::TxConfig.load_file(project_config['tx_config'])
-        Txgh::Config.new(project_config, repo_config, tx_config)
+        Txgh::Config.new(project_config, repo_config)
       end
 
-      def config_from(project_name, repo_name, tx_config = nil)
+      def config_from(project_name, repo_name)
         project_config = project_config_for(project_name)
         repo_config = repo_config_for(repo_name)
-        tx_config ||= Txgh::TxConfig.load_file(project_config['tx_config'])
-        Txgh::Config.new(project_config, repo_config, tx_config)
+        Txgh::Config.new(project_config, repo_config)
+      end
+
+      def tx_config(transifex_project, github_repo, ref = nil)
+        scheme, payload = split_uri(transifex_project.tx_config_uri)
+
+        case scheme
+          when 'raw'
+            Txgh::TxConfig.load(payload)
+          when 'file'
+            Txgh::TxConfig.load_file(payload)
+          when 'git'
+            unless ref
+              raise TxghError,
+                "TX_CONFIG specified a file from git but did not provide a ref."
+            end
+
+            Txgh::TxConfig.load(
+              github_repo.api.download(github_repo.name, payload, ref)
+            )
+        end
       end
 
       private :new
 
       private
 
-      def yaml
-        path = if File.file?(File.join(Etc.getpwuid.dir, "txgh.yml"))
-          File.join(Etc.getpwuid.dir, "txgh.yml")
+      def split_uri(uri)
+        if uri =~ /\A[\w]+:\/\//
```

Comment:
can the URI lib help here?
http://ruby-doc.org/stdlib-2.1.1/libdoc/uri/rdoc/URI.html